### PR TITLE
Allow removing DEFINER clause from exported SQL

### PR DESCRIPTION
**Pull request by erolatreston**
### Description

Allow removing the DEFINER clause from the exported event, view and routine definitions.

Before submitting pull request, please review the following checklist:

- [ ] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [ ] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [ ] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [ ] Every commit has a descriptive commit message.
- [ ] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [ ] Any new functionality is covered by tests.

### CODE REVIEW

File: libraries/classes/Plugins/Export/ExportSql.php [hunk 1]
```diff
@@ -561,6 +561,10 @@ protected function exportRoutineSQL(
                 '',
                 $flag
             );
+            if ($GLOBALS['cfg']['Export']['removeDefinerFromDefinitions']) {
+                // Remove definer clause from routine definitions
+                $createQuery = preg_replace('/ DEFINER=`[a-zA-Z0-9_\.]*`@`.*?` /', ' ', $createQuery);
```
Comment (williamdes) [inline libraries/classes/Plugins/Export/ExportSql.php:566]: we should investigate if this can be done with the sql-parser lib

**Review by williamdes — APPROVED** [review]
Looks good to me

File: libraries/classes/Plugins/Export/ExportSql.php (hunk 1 above)
Comment (erolatreston) [reply libraries/classes/Plugins/Export/ExportSql.php]: Now using the parser to remove the DEFINER.

**Review by williamdes — APPROVED** [review]
💯 Awesome work !

**Review by MauricioFauth — APPROVED** [review]

**Comment by mauriciofauth** [comment]
Merged, thanks for your contribution!

**Pull request merged**